--- lib/Bric/Biz/OutputChannel/Element.pm.~1.2.4.1.~	Mon Jun  9 19:30:08 2003
+++ lib/Bric/Biz/OutputChannel/Element.pm	Fri Jun 13 09:10:48 2003
@@ -78,8 +78,11 @@
   ', eoc.id, eoc.element__id, eoc.enabled';
 my @SEL_PROPS = (Bric::Biz::OutputChannel::SEL_PROPS(),
                  qw(_map_id element_id _enabled));
-my $SEL_TABLES = Bric::Biz::OutputChannel::SEL_TABLES() .
-  ', element__output_channel eoc';
+
+# Grabbed knowledge from parent, but the outer join depends on it. :-(
+my $SEL_TABLES = 'output_channel oc LEFT OUTER JOIN ' .
+  'element__output_channel eoc ON (oc.id = eoc.output_channel__id), ' .
+  'member m, output_channel_member sm';
 
 sub SEL_PROPS { @SEL_PROPS }
 sub SEL_COLS { $SEL_COLS }
@@ -201,6 +204,8 @@
     }
     # Set the necessary properties and return.
     $self->_set([qw(_enabled element_id)], [$en, $eid]);
+    # New relationships should always trigger a save.
+    $self->_set__dirty(1);
 }
 
 ##############################################################################
